### PR TITLE
Tian boon - Implemented base64 Encoder/Decoder classes

### LINKED ISSUE #18: Add Base64Encoder Class

### LINKED ISSUE #19: Add Base64Decoder Class

### PR DESCRIPTION
**Pull request by Kair0s3**
Fixes #18 
Fixes #19

### CODE REVIEW

File: src/main/java/decodex/modules/Base64Decoder.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.Base64;
+
+import decodex.data.Data;
+
+public class Base64Decoder extends BaseModule {
+
+    private final Base64.Decoder base64Decoder;
+
+    public Base64Decoder(String name, String description) {
```
Comment (arraysius) [inline src/main/java/decodex/modules/Base64Decoder.java:11]: Module name and description is constant. I don't think the constructor needs to take in parameters. Similar with Base64Encoder.

**Review by arraysius — COMMENTED** [review]
Looks good for the most part. Only the constructors raise questions.

File: src/main/java/decodex/modules/Base64Encoder.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package decodex.modules;
+
+import java.util.Base64;
+
+import decodex.data.Data;
+
+public class Base64Encoder extends BaseModule {
```
Comment (rizemon) [inline src/main/java/decodex/modules/Base64Encoder.java:7]: Perhaps add a simple comment here to describe the class?

**Review by rizemon — COMMENTED** [review]
A few changes to be made, otherwise looks good to me.

File: src/test/java/decodex/modules/Base64DecoderTest.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    void run_encodedString_base64DecodedData() {
+        Base64Decoder base64Decoder = new Base64Decoder();
+        String encodedString = "dGVzdA==";
+        String expectedString = "test";
+
+        Data encodedData = new Data(encodedString);
+        Data decodedData = base64Decoder.run(encodedData);
+        assertEquals(expectedString, decodedData.toString());
+    }
+}
```
Comment (rizemon) [inline src/test/java/decodex/modules/Base64DecoderTest.java:20]: Remember to add a blank line at the end of all files (Git convention)

**Review by rizemon — CHANGES_REQUESTED** [review]
Just add the extra new lines. Other than that, looks good to me.

**Pull request merged**